### DISCUSSION TITLE
Decrypting with passphrase issue

### DISCUSSION THREAD

**Discussion by Maxduck** [FiloSottile/age#447] [Q&A]
Hi... Hope this is an appropriate place to ask...

This stems from using the age encyption on Logseq.

Short story is that my files were encrypted and then I used [https://github.com/kanru/logseq-encrypt-ui](url) to decrypt the files.

This left me with some empty files and no way to decrypt the originals (wherever the key file was kept seems to have been written over).

If I try and decrpyt them I get the following error:
`age: error: no identity matched any of the recipients`

Is there a way I can recover the files? (no answers yet at logseq-encrypt-ui)

Thanks!

**Comment by itsamemarkus** [comment]
Hi, I don't know if you're the same person that just asked in logseq discord, but that was a nice puzzle for the evening.

I haven't scripted the automation to decrypt everything automatically, but to decrypt pages manually you first have to get the age secret key from `logseq/metadata.edn` with your password also known as age passphrase (finding that was the hardest part ^^)

I made a video for the whole process:

https://user-images.githubusercontent.com/34944988/189221240-40cac9ed-dcbb-4c6b-a112-9d04e87d3b74.mp4

The secret key you get there is valid for every encrypted file then.

In case the `logseq-encrypt-ui` tool messed something up and `metadata.edn` got overwritten because you encrypted and decrypted multiple times you might be out of luck. I tested it and it generated a different age secret even if you enter the same password.

Hope this works for you.

**Reply by Maxduck** [reply]
Hey @itsamemarkus, thanks so much for responding and taking the time to make the video!

Unfortunately, although I only used the `logseq-encrypt-ui` tool once, the `metadata.edn` was overwritten.... so I did lose the original age secret key which would decrypt the files... :(

I tried regenerating the key with the same passcode... but didn't work, so it makes sense that you said it generated a new age secret key.

Thanks again... and hope you enjoyed your evening puzzle!
